### PR TITLE
Locale Handling Improvements

### LINKED ISSUE #342: Sprache basierend auf Browsersprache wählen
Als Nutzer möchte ich,
dass die Sprache der Webseite automatisch an die meines Browsers angepasst wird, sofern ich sie nicht manuell ändere,
damit ich nicht bei jedem Besuch händisch von Englisch auf Deutsch wechseln muss.

### PR DESCRIPTION
**Pull request by sebschrader**
This pull request improves the locale handling in SIPA quite significantly. In particular it addresses locale negotiation and locale preference storage.

1. The locale is now negotiated with Babel's `negotiate_locale` function as outlined in #344, which will fix #342.

2. The `localeselector` is no longer overloaded to handle three different concerns (obtaining the user personal preference, persisting the preference in the session and actually selecting a locale). It only does one thing now: Selecting a locale. Obtaining and saving the locale is now done in a separate functions. `get_user_locale_preference` and `save_user_locale_preference`. This design makes eventually implementing #233 much easier.

3. The separate `set_language` endpoint is removed. It was never needed, because the special request query parameter `lang` has always been evaluated. This query parameter was renamed to `locale`.

4. The `locale` value is no longer stored in the `session` object. It is stored in a separate cookie (configurable and named `locale` by default) with a different lifetime (persistent and 31 days). This  addresses #292 (what also #294 aimed to do). It can't fix it completely unfortunately, `Flask-Login` is still causing a session cookie to be set, due to its [Session Protection feature](https://flask-login.readthedocs.io/en/latest/#session-protection), which sets a `_fresh` key in the session.

### CODE REVIEW

File: sipa/initialization.py [hunk 1]
```diff
@@ -145,7 +151,7 @@ def try_register_uwsgi_timer(app):
         @timer(300)
         def update_uwsgi(signum):
             flatpages_root = app.config["FLATPAGES_ROOT"]
-            logger.debug("Updating git repository at %s", flatpages_root)
+            logger.debug("Udpating git repository at %s", flatpages_root)
```
Comment (lukasjuhrich) [inline sipa/initialization.py]: That appears superfluous tho ;)

Comment (mmarx) [reply sipa/initialization.py]: It even introduces a typo.

Comment (sebschrader) [reply sipa/initialization.py]: Rebase fail

**Comment by lukasjuhrich** [comment]
@sebschrader Sorry that I didn't manage to merge this PR within October. I will try to get back to it until the next coding session on wednesday.

**Pull request merged**